### DISCUSSION TITLE
Build error

### DISCUSSION THREAD

**Discussion by jdlee0117** [bitcraze/discussions#296] [Q&A]
Hi,
I am trying to build the firmware on my Mac based on the instructions in:
[Building and flashing](urhttps://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#compilingl)

However, I am experiencing the following error:
*** No rule to make target 'vendor/CMSIS/CMSIS/DSP/Source/BasicMathFunctions/arm_abs_f32.o', needed by 'vendor/built-in.o'.  Stop.
*** [tools/kbuild/Makefile.kbuild:589: vendor] Error 2

Any suggestions how to solve this issue?

Thanks

**Comment by krichardsson** [comment]
Have you [cloned the repo recursively](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#cloning)? 

try 
``` bash
$ cd crazyflie-firmware
$ git submodule init
$ git submodule update
```

**Reply by jdlee0117** [reply]
Thank you very much for your feedback. While I was trying "git submodule update", I got the following error:

Aborting
Unable to checkout 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4' in submodule path 'vendor/CMSIS'

How do I fix this?

Thank you.

**Reply by knmcguire** [reply]
strange... have you tried doing a `git submodule sync` and then a 'git submodule update' ?

**Comment by jdlee0117** [comment]
Hi Kimberly,

Sorry for the late response.

Yes, I did and this is what I got... (please see attached picture).

Best,
Leo


________________________________

**Reply by knmcguire** [reply]
Hi! If you replied by email, we can not see the attached picture in github. Go to the actual discussion patch and attach the picture there. 

Could you also try to just make the repo at a different location by git cloning it fresh:
`git clone --recursive https://github.com/bitcraze/crazyflie-firmware.git
`

**Reply by jdlee0117** [reply]
Hi, thank you very very much for your patience on helping me solving this compiling issue.

> Could you also try to just make the repo at a different location by git cloning it fresh:
Yes, I tried to make the repo on my Desktop, but still got the same issue (please see the attached picture)

![git clone issue](https://user-images.githubusercontent.com/12797069/195374503-301568b6-367a-4929-9e41-3157ed057ac6.png)

Your help appreciated!!!

**Reply by krichardsson** [reply]
This is hard for us to solve, it seems to be a git problem and it is maybe better to look for solutions in a git forum.

I just tested to run `git clone --recursive https://github.com/bitcraze/crazyflie-firmware.git` and it worked fine on my machine.

The suggestions I have is to upgrade git and make sure your git environment works.
